### PR TITLE
ShyLU_DDD/FROSch: Some fixes, mostly related to the TwoLevelBlockPreconditioner

### PR DESCRIPTION
**Pull request by searhein**
@trilinos/shylu 

## Motivation

While working on the monolithic preconditioner, that is, the `TwoLevelBlockPreconditioner`, I have seen several issues which either prevented the use of certain parameters from the parameterlist or even caused segmentation faults when called through the Stratimikos interface.

Moreover, this PR fixes two other small issues:
+ While linking `FROSch` from an external library, we have seen that some type definitions were missing when including `FROSch_SchwarzPreconditioner_decl.hpp`.
+ Some mistakes in the output.

These issues have been fixed.

## Testing

All `FROSch` tests in Trilinos as well as the tests in the external library pass.

### CODE REVIEW

File: packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_IPOUHarmonicCoarseOperator_def.hpp [hunk 1]
```diff
@@ -218,16 +218,36 @@ namespace FROSch {
...
+        FROSCH_ASSERT(dofsPerNodeVec.size()==this->NumberOfBlocks_,"dofsPerNodeVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(repeatedDofMapsVec.size()==this->NumberOfBlocks_,"repeatedDofMapsVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(nullSpaceBasisVec.size()==this->NumberOfBlocks_,"nullSpaceBasisVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(dirichletBoundaryDofsVec.size()==this->NumberOfBlocks_,"dirichletBoundaryDofsVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(nodeListVec.size()==this->NumberOfBlocks_,"nodeListVec.size()!=this->NumberOfBlocks_");
+
         // Das könnte man noch ändern
         // TODO: DAS SOLLTE ALLES IN EINE FUNKTION IN HARMONICCOARSEOPERATOR
-        for (UN i=0; i<repeatedNodesMapVec.size(); i++) {
+        for (UN i=0; i<this->NumberOfBlocks_; i++) {
```
Comment (srajama1) [inline packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_IPOUHarmonicCoarseOperator_def.hpp:232]: Help us folks with no German knowledge in the comments please :) .. It is pretty cool Trilinos has comments in several languages though

**Review by srajama1 — DISMISSED** [review]
Minor comment below.

File: packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_IPOUHarmonicCoarseOperator_def.hpp (hunk 1 above)
Comment (searhein) [reply packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_IPOUHarmonicCoarseOperator_def.hpp:231]: Haha, you are right. I should pay more attention to this. I translated this now :-)

**Review by roeverf — APPROVED** [review]
Thanks for the changes!

**Pull request merged**